internal/policy/policy.go

```diff
@@ -36,21 +36,54 @@ import (
 	"github.com/hacbs-contract/ec-cli/internal/kubernetes"
 )
 
-type Policy struct {
+type Policy interface {
+	PublicKeyPEM() ([]byte, error)
+	CheckOpts() (*cosign.CheckOpts, error)
+	WithSpec(spec ecc.EnterpriseContractPolicySpec) Policy
+	Spec() ecc.EnterpriseContractPolicySpec
+	EffectiveTime() time.Time
+}
+
+type policy struct {
 	ecc.EnterpriseContractPolicySpec
```

Comment:
Should this be `spec ecc.EnterpriseContractPolicySpec` now? Not sure if there's any value in "extending" that type.

Reply:
We break backward compatibility when `--policy` is an inline policy otherwise; less we copy all the fields from `EnterpriseContractPolicySpec` into `policy`.

Reply:
Yeah... We could add something like `NewFromECPSpec`. Probably not worth doing it at this time.